core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -566,8 +558,8 @@ protected void doRun() {
             }
 
             // resolve all derived request fields, so we can route and apply it
-            if (request.consistencyLevel() == WriteConsistencyLevel.DEFAULT) {
-                request.consistencyLevel(defaultWriteConsistencyLevel);
+            if (request.waitForActiveShards() == ActiveShardCount.DEFAULT) {
+                request.waitForActiveShards(WAIT_FOR_ACTIVE_SHARDS_SETTING.get(indexMetaData.getSettings()));
```

Comment:
hmm... that does a map lookup on every op, not got. Can we have the previous variable back and add a setting listener that updates it?